api/pkg/apis/v1alpha1/providers/stage/wait/wait.go

```diff
@@ -301,6 +301,20 @@ func (i *WaitStageProvider) Process(ctx context.Context, mgrContext contexts.Man
 			return outputs, false, nil
 		}
 		counter++
+		if i.Config.WaitCount == 0 && counter%10 == 0 {
+			// Avoid to check the activation status too frequently
+			activationName, ok := inputs["__activation"].(string)
+			if !ok || activationName == "" {
+				log.InfoCtx(ctx, "  P (Wait Processor): __activation is not provided, exiting endless wait.")
+				return outputs, false, nil
+			}
+			_, err := i.ApiClient.GetActivation(ctx, activationName, namespace, i.Config.User, i.Config.Password)
+			if err != nil {
```

Comment:
You need to check if error type is NotFound or not. We should not break the loop for other transient errors.